app/assets/javascripts/directives/hostname_validation.js

```diff
@@ -0,0 +1,33 @@
+(function() {
+  angular.module('ManageIQ').directive('hostnameValidation', directive);
+
+  function directive() {
+    return {
+      require: 'ngModel',
+      restrict: 'A',
+      link: link,
+    };
+    function link(_scope, _elem, _attrs, ctrl) {
+      ctrl.$validators.hostnameValidation = function(_modelValue, viewValue) {
+        return ! viewValue || validateHostname(viewValue);
+      };
+      /**
+       * The hostname is invalid if:
+       *  - Has protocol before the address (ex: http://);
+       *  - If it is an IP address and has some path after the address;
+       * @param {any} hostname - Hostname to be validate
+       * @returns {boolean} True if the hostname is valid
+       */
+      var validateHostname = function(hostname) {
+        return ! hasProtocol(hostname) && ! isIpAddressWithPath(hostname);
+      };
+      var hasProtocol = function(hostname) {
+        return /\b:\/\/\b/.test(hostname);
+      };
+      var isIpAddressWithPath = function(hostname) {
+        var isIpAddress = /\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b/.test(hostname);
```

Comment:
I think the following cases should be invalid: `100.100.100.100invalid`, `invalid100.100.100.100invalid`. Could you confirm that?

Reply:
In addition, this does not support IPV6 format, pointed out in the UI.